modules/hangman.js

```diff
@@ -1,150 +1,294 @@
-        this.commands = {
-            hangman: {
-                aliases: [],
-                inline: false,
-                description: 'Start a game of hangman, where you have to guess the card name with reaction letters',
-                help: 'Selects a random Magic card, token oder plane and shows you the ' +
-                'placeholders for each letter in its name. To guess a letter, add a reaction to the ' +
-                'Hangman-message of the bot. Reactions have to be selected among the regional indicators ' +
-                ':regional_indicator_a: to :regional_indicator_z:. You can only have one active game of ' +
-                'Hangman per Discord server, but you can also start an additional one in a private query.\n\n' +
-                '**Difficulty**:\n' +
-                'With an optional parameter you can select the difficulty. Available are `easy`, `medium` ' +
-                'and `hard`. Default is medium. "Easy" includes the mana cost and type line, "Medium" includes ' +
-                'the mana cost and "Hard" doesn\'t include any hints.',
-                examples: ["!hangman", "!hangman easy"]
-            }
-        };
-        this.cardApi = "https://api.scryfall.com/cards/random";
-        this.gameTime = 3*60*1000; // minutes
-        this.runningGames = [];
+const GAME_TIME = 3 * 60 * 1000; // minutes
+
+class HangmanGame {
+    constructor({
+                    id,
+                    gameList,
+                    message = null,
+                    collector = null,
+                    card = null,
+                    difficulty = 'medium',
+                } = {}) {
+        this.id = id;
+        this.gameList = gameList;
+        this.message = message;
+        this.collector = collector;
+        this.card = card;
+        this.difficulty = difficulty;
+
+        this.letters = new Set();
+        this.wrongGuesses = 0;
+        this.done = false;
+        this.gameSuccess = false;
     }
 
-    getCommands() {
-        return this.commands;
+    /**
+     * Handle a user guessing a letter in the card name
+     * @param {String} char The character that was guessed
+     */
+    handleLetter(char) {
+        // Letters is a set, which handles duplicates for us
+        this.letters.add(char);
+        this.checkDone();
+        this.updateEmbed(false);
     }
 
-    // generate the embed card
-    generateEmbed(card, difficulty, letters = [], done = false) {
-        // count number of wrong letters and missing letters
-        const wrong = letters.filter(c => card.name.toLowerCase().indexOf(c) === -1).length;
-        const missing = _.difference(_.uniq(card.name.replace(/[^a-z]/ig,'').toLowerCase().split("")), letters);
+    /**
+     * Handle a user guessing the entire card name
+     * @param {String} guess
+     * @param {Discord.Message} msg
+     */
+    handleGuess(guess, msg) {
+        const correct = this.card.name.toLowerCase();
+        if (guess.includes(correct)) {
```

Comment:
Currently, we don't strip out special characters, which makes guessing a bit more tedious since those characters are already shown anyway. Maybe we should strip out those before comparing in a future release.

Reply:
Fair point